### PR TITLE
IOS-190 Access property via accessor to ensure consistency

### PR DESCRIPTION
**Pull request by ettore**
**What's this do?**
this crash is caused by an unrecognized message sent to an invalid `DefaultAudiobookManager` instance. While the class has some possible threading issues, I don't think they are the cause of this crash, because the incriminated `manager` property is being deallocated as part of the timer behavior (which appears correct to me). However, at the point of crash we were accessing the lower level pointer instead of the property, and the property was not `atomic`, so it is possible (depending on the optimizations done by the runtime) to access a garbage pointer. By using `atomic` and the property we should avoid this situation.

If the crash persists, I think it will be time (no pun intended 😄 ) for a larger refactor (ensuring thread safety, perhaps avoiding the timer entirely) that I don't have bandwidth for right now.

**Why are we doing this? (w/ JIRA link if applicable)**
https://jira.nypl.org/browse/IOS-190

**How should this be tested? / Do these changes have associated tests?**
will add in ticket

**Dependencies for merging? Releasing to production?**
no, this will go out after 3.6.5

**Does this include changes that require a new SimplyE build for QA?**
not needed right now

**Has the application documentation been updated for these changes?**
n/a

**Did someone actually run this code to verify it works?**
@ettore

### CODE REVIEW

File: Simplified/Book/UI/NYPLBookCellDelegate.m [hunk 1]
```diff
@@ -330,12 +331,14 @@ - (void)scheduleTimerForAudiobook:(NYPLBook *)book
...
-                                              selector:@selector(pollAudiobookReadingLocation)
+                                              selector:@selector(pollAudiobookReadingLocation:)
                                               userInfo:nil
                                                repeats:YES];
 }
 
-- (void)pollAudiobookReadingLocation
+// non-thread safe: must be called on the same thread as
+// scheduleTimerForAudiobook:manager:viewController: each time.
+- (void)pollAudiobookReadingLocation:(NSTimer *)timer
```
Comment (ErnestFan) [inline Simplified/Book/UI/NYPLBookCellDelegate.m:341]: Does adding the `NSTimer` as the parameter do anything here? Just curious

**Review by ErnestFan — APPROVED** [review]

File: Simplified/Book/UI/NYPLBookCellDelegate.m (hunk 1 above)
Comment (ettore) [reply Simplified/Book/UI/NYPLBookCellDelegate.m:341]: no, it does nothing... I merely added it to comply with the docs. But it worked without too.

File: Simplified/Book/UI/NYPLBookCellDelegate.m [hunk 2]
```diff
@@ -30,7 +30,7 @@ @interface NYPLBookCellDelegate () <RefreshDelegate>
 
 @property (nonatomic) NSTimer *timer;
 @property (nonatomic) NYPLBook *book;
-@property (nonatomic) id<AudiobookManager> manager;
+@property DefaultAudiobookManager *manager;
```
Comment (singhraman4282) [inline Simplified/Book/UI/NYPLBookCellDelegate.m:33]: There's a huge change that I'm missing something here but it looks like we're breaking the dependency inversion principle.

Comment (ettore) [reply Simplified/Book/UI/NYPLBookCellDelegate.m:33]: My changes are pretty minimal, so I'm guessing you mean with the existing implementation?

Comment (singhraman4282) [reply Simplified/Book/UI/NYPLBookCellDelegate.m:33]: oops sorrry I meant CHANCE not change @ettore

**Review by singhraman4282 — APPROVED** [review]

**Review by singhraman4282 — APPROVED** [review]

File: Simplified/Book/UI/NYPLBookCellDelegate.m (hunk 2 above)
Comment (ettore) [reply Simplified/Book/UI/NYPLBookCellDelegate.m:33]: hey Raman can you explain what you mean? this is for a private ("private" in objc terms that is) property so calling things with their actual name is not making things more tightly dependent than using a protocol.

Comment (singhraman4282) [reply Simplified/Book/UI/NYPLBookCellDelegate.m:33]: you're right. I missed the 'private' part.

**Pull request merged**